### ISSUE TITLE
Make Optimzer take adaptive learning rate

### ISSUE THREAD

**Issue by icebreaker2** [icebreaker2/AI#36]
Use 
* Tf callbacks or at the optimizers?!?
* other gelators
  * weight decay
  * learning rate decay based on epoch

**Get to know others!!!!**
